src/common/containers/build-details.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import Helmet from 'react-helmet';
+
+import BuildRow from '../components/build-row';
+import BuildLog from '../components/build-log';
+import { fetchBuilds } from '../actions/snap-builds';
+
+import styles from './container.css';
+
+class BuildDetails extends Component {
+
+  componentWillMount() {
+    this.props.dispatch(fetchBuilds(this.props.fullName));
+  }
+
+  render() {
+    const { account, repo, fullName, buildId, build } = this.props;
+
+    return (
+      <div className={ styles.container }>
+        <Helmet
+          title={`${fullName} builds`}
+        />
+        {/* TODO: make into title component? */}
+        <h1>{fullName} build #{buildId}</h1>
+        { this.props.isFetching &&
+          <span>Loading...</span>
+        }
+        { build &&
+          <div>
+            <BuildRow account={account} repo={repo} {...build} />
+            <h3>Build log:</h3>
+            <BuildLog logUrl={build.buildLogUrl} />
+          </div>
+        }
+      </div>
+    );
+  }
+
+}
+BuildDetails.propTypes = {
+  account: PropTypes.string.isRequired,
+  repo: PropTypes.string.isRequired,
+  fullName: PropTypes.string.isRequired,
+  buildId: PropTypes.string.isRequired,
+  build: PropTypes.object,
+  isFetching: PropTypes.bool,
+  dispatch: PropTypes.func.isRequired
+};
+
+const mapStateToProps = (state, ownProps) => {
+  const account = ownProps.params.account.toLowerCase();
+  const repo = ownProps.params.repo.toLowerCase();
+  const buildId = ownProps.params.buildId.toLowerCase();
+
+  const fullName = `${account}/${repo}`;
+  // TODO:
+  // find build by id in builds list
+  // but it should also fetch it if necessary
+  const build = state.snapBuilds.builds.filter((build) => build.buildId === buildId)[0];
```

Comment:
Very basic 'find by id' implementation to find specific build details in list of the builds.

Currently it works on assumption that we have a store with list of builds of given repository. It gets bit tricky when user goes directly to details page (when listing was not yet fetched). Now, with mocked data, I just 'fetch' whole mocked list of builds, but in the end we should probably only request single builds details and fetch listing only when user goes to listing page.